Framework/API/src/ExperimentInfo.cpp

```diff
@@ -79,6 +79,15 @@ ExperimentInfo::ExperimentInfo(const ExperimentInfo &source) {
   setSpectrumDefinitions(source.spectrumInfo().sharedSpectrumDefinitions());
 }
 
+/**
+ * Implements the copy assignment operator
+ */
+ExperimentInfo &ExperimentInfo::operator=(const ExperimentInfo &source) {
```

Comment:
May I suggest that we use the ["copy-and-swap"](https://stackoverflow.com/a/3652138) idiom to implement the copy constructor/assignment operator pair here? It is a nice idiom for reducing code duplication and providing exception guarantees at the same time.